csharp/test/Apache.Arrow.Adbc.Tests/Metadata/GetObjectsParser.cs

```diff
@@ -173,6 +173,14 @@ private static List<AdbcConstraint> ParseConstraints(StructArray constraintsArra
                 StringArray col_names = column_names.GetSlicedValues(i) as StringArray;
                 StructArray usage = column_usage.GetSlicedValues(i) as StructArray;
 
+                if (col_names != null)
+                {
+                    for (int j = 0; j < col_names.Length; j++)
```

Comment:
We should reserve a capacity on the target list since the number of columns can be large.

Reply:
Here the columns are for composite keys. Though Snowflake doesn't set a limit for it, most other SQL databases set a limit of 16 to 32 on composite indexes. It is 16 for [MySQL](https://dev.mysql.com/doc/refman/8.0/en/multiple-column-indexes.html#:~:text=10.3.6%C2%A0Multiple%2DColumn,Column%20Indexes%E2%80%9D). For most practical cases, it would be 1-2 columns. Reserving capacity for it may give minimal performance gains. What do you suggest?